guides/bulk.md

```diff
@@ -0,0 +1,178 @@
+
+# Provide a CA bundle if you use intermediate CAs with your root CA.
+# If this is not given, the CA bundle is is discovered from the first available
+# following options:
+# - OpenSSL environment variables SSL_CERT_FILE and SSL_CERT_DIR
+# - certifi bundle (https://pypi.org/project/certifi/)
+# - default behavior of the connection backend (most likely system certs)
+ca_certs_path = '/full/path/to/root-ca.pem'
+
+# Optional client certificates if you don't want to use HTTP basic authentication.
+# client_cert_path = '/full/path/to/client.pem'
+# client_key_path = '/full/path/to/client-key.pem'
+
+# Create the client with SSL/TLS enabled, but hostname verification disabled.
+client = OpenSearch(
+    hosts = [{'host': host, 'port': port}],
+    http_compress = True, # enables gzip compression for request bodies
+    http_auth = auth,
+    # client_cert = client_cert_path,
+    # client_key = client_key_path,
+    use_ssl = True,
+    verify_certs = True,
+    ssl_assert_hostname = False,
+    ssl_show_warn = False,
+    ca_certs = ca_certs_path
+)
+```
+
+Next, create an index named `movies` and another named `books` with the default settings:
+
+```python
+movies = "movies"
+books = "books"
+client.indices.create(movies) if not client.indices.exists(movies)
+client.indices.create(books) if not client.indices.exists(books)
+```
+
+## Bulk API
+
+The `bulk` API action allows you to perform document operations in a single request. The body of the request is an array of objects that contains the bulk operations and the target documents to index, create, update, or delete.
+
+### Indexing multiple documents
+
+The following code creates two documents in the `movies` index and one document in the `books` index:
+
+```python
+client.bulk(
+  [
+    { "index": { "_index": "movies", "_id": "1" } },
+    { "title": "Beauty and the Beast", "year": "1991" },
+    { "index": { "_index": "movies", "_id": "2" } },
+    { "title": "Beauty and the Beast - Live Action", "year": "2017" },
+    { "index": { "_index": "books", "_id": "1" } },
+    { "title": "The Lion King", "year": "1994" }
+  ]
+)
+```
+
+As you can see, each bulk operation is comprised of two objects. The first object contains the operation type and the target document's `_index` and `_id`. The second object contains the document's data. As a result, the body of the request above contains six objects for three index actions.
+
+Alternatively, the `bulk` method can accept an array of hashes where each hash represents a single operation. The following code is equivalent to the previous example:
+
+```python
+client.bulk(
+  [
+    { "index": { "_index": "movies", "_id": "1", "data": { "title": "Beauty and the Beast", "year": "1991" }  } },
+    { "index": { "_index": "movies", "_id": "2", "data": { "title": "Beauty and the Beast - Live Action", "year": "2017" } } },
+    { "index": { "_index": "books", "_id": "1", "data": { "title": "The Lion King", "year": "1994" } } }
+  ]
+)
+```
```

Comment:
Please correct me if I am wrong, above code is giving illegal_argument_exception. I tried with opensearch-py 2.2.0